### PR TITLE
docs: Corrections and edits on release process document

### PR DESCRIPTION
**Pull request by terrytangyuan**
These edits/corrections are found useful during the release process https://github.com/kserve/kserve/pull/3325.

### CODE REVIEW

File: release/RELEASE_PROCESS_v2.md [hunk 1]
```diff
@@ -77,17 +77,17 @@ You can repeat same steps for RC2 or other release candidates if needed
...
-   4. [Chart.yaml in kserve-resources](../charts/kserve-resources/Chart.yaml#L3) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_VERSION}`
-   5. [values.yaml in kserve-resources](../charts/kserve-resources/values.yaml#L2) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_VERSION}`
-2. Add a new version `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_VERSION}` in the `RELEASES` array in [generate-install.sh](../hack/generate-install.sh). Example: Refer [this commit](https://github.com/rachitchauhan43/kserve/commit/6e9bd24ea137a3619da3297b4ff000379f7b2b38#diff-5f8f3e3a8ca601067664c7bf00c05aa2290a6ba625312754856ec873b840b6dbR42)
+   1. [VERSION](../python/VERSION) to `${MAJOR}.${MINOR}.${PATCH}`
+   2. [quick_install.sh](../hack/quick_install.sh#L36) to `v${MAJOR}.${MINOR}.${PATCH}`
+   3. [Chart.yaml in kserve-crd](../charts/kserve-crd/Chart.yaml#L3) to `v${MAJOR}.${MINOR}.${PATCH}`
+   4. [Chart.yaml in kserve-resources](../charts/kserve-resources/Chart.yaml#L3) to `v${MAJOR}.${MINOR}.${PATCH}`
+   5. [values.yaml in kserve-resources](../charts/kserve-resources/values.yaml#L2) to `v${MAJOR}.${MINOR}.${PATCH}`
```
Comment (terrytangyuan) [inline release/RELEASE_PROCESS_v2.md:85]: Correct me if I am wrong but I think in this section, we are releasing the GA version, not the release candidate any more.

File: release/RELEASE_PROCESS_v2.md [hunk 2]
```diff
@@ -77,17 +77,17 @@ You can repeat same steps for RC2 or other release candidates if needed
...
+2. Add a new version `v${MAJOR}.${MINOR}.${PATCH}` in the `RELEASES` array in [generate-install.sh](../hack/generate-install.sh). Example: Refer [this commit](https://github.com/rachitchauhan43/kserve/commit/6e9bd24ea137a3619da3297b4ff000379f7b2b38#diff-5f8f3e3a8ca601067664c7bf00c05aa2290a6ba625312754856ec873b840b6dbR42)
 3. Generate install manifest `./hack/generate-install.sh $VERSION`.
 4. Run `./hack/python-release.sh` to update pyproject.toml files for all packages.
```
Comment (terrytangyuan) [inline release/RELEASE_PROCESS_v2.md]: Is it intentional to only run this for GA release but not RCs?

Comment (terrytangyuan) [reply release/RELEASE_PROCESS_v2.md]: This also needs to be run for RCs. See https://github.com/kserve/kserve/pull/3325#issuecomment-1865513404

File: release/RELEASE_PROCESS_v2.md [hunk 3]
```diff
@@ -40,21 +40,22 @@ We will be creating the first release candidate (RC0) on the feature freeze day
...
-    1. [VERSION](../python/VERSION) to `${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`
+    1. [VERSION](../python/VERSION) to `${MAJOR}.${MINOR}.${PATCH}rc${RELEASE_CANDIDATE_VERSION}` (note that the dash before "rc" is removed intentionally to meet [PyPI package version requirements](https://packaging.python.org/en/latest/specifications/version-specifiers/#public-version-identifiers))
     2. [quick_install.sh](../hack/quick_install.sh#L36) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`
     3. [Chart.yaml in kserve-crd](../charts/kserve-crd/Chart.yaml#L3) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`
     4. [Chart.yaml in kserve-resources](../charts/kserve-resources/Chart.yaml#L3) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`
-    5. [values.yaml in kserve-resources](../charts/kserve-resources/values.yaml#L2) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`
+    5. [values.yaml in kserve-resources](../charts/kserve-resources/values.yaml#L2) to `v${MAJOR}.${MINOR}.${PATCH}-rc${RELEASE_CANDIDATE_VERSION}`. In addition, bump `modelmeshVersion` if it a newer version of ModelMesh is available.
```
Comment (terrytangyuan) [inline release/RELEASE_PROCESS_v2.md:47]: Added this

**Comment by terrytangyuan** [comment]
This is ready for review. Most of the changes are exercised during v0.12 release https://github.com/kserve/kserve/pull/3325

/cc @yuzisun

File: release/RELEASE_PROCESS_v2.md (hunk 2 above)
Comment (yuzisun) [reply release/RELEASE_PROCESS_v2.md]: yes, this should be done for RC as well

**Pull request merged**